salt/modules/nilrt_ip.py

```diff
@@ -645,23 +646,10 @@ def _change_state(interface, new_state):
     '''
     if __grains__['lsb_distrib_id'] == 'nilrt':
         return _change_state_legacy(interface, new_state)
-    service = _interface_to_service(interface)
-    if not service:
-        if interface in map(lambda x: x.name, pyiface.getIfaces()):
-            ret = _change_connman_backlist(interface, add=new_state == 'down')
-            return ret and __salt__['cmd.run_all']('/etc/init.d/connman restart')['retcode'] == 0
-        raise salt.exceptions.CommandExecutionError('Invalid interface name: {0}'.format(interface))
-    connected = _connected(service)
-    if (not connected and new_state == 'up') or (connected and new_state == 'down'):
-        service = pyconnman.ConnService(os.path.join(SERVICE_PATH, service))
-        try:
-            state = service.connect() if new_state == 'up' else service.disconnect()
-            _change_connman_backlist(interface, add=new_state == 'down')
-            return state is None
-        except Exception:
-            raise salt.exceptions.CommandExecutionError('Couldn\'t {0} service: {1}\n'
-                                                        .format('enable' if new_state == 'up' else 'disable', service))
-    return True
+    if interface in map(lambda x: x.name, pyiface.getIfaces()):
+        ret = _change_connman_blacklist(interface, add=new_state == 'down')
```

Comment:
add=new_state == 'down' looks confusing. You could either get that an a different line, or add parentheses

Reply:
In addition, there should be some tests to accompany this change.